som/radius.go

```diff
@@ -0,0 +1,26 @@
+package som
+
+import "math"
+
+// Radius is a decay function for the radius parameter.  The supported strategies
+// are "exp" and "lin".  "exp" is an exponential decay function, "lin" is linear.
+// At iteration 0 the function returns the radius0, at totalIterations it returns 1.0
+func Radius(iteration, totalIterations int, strategy string, radius0 float64) float64 {
```

Comment:
What if someone passes in negative number of `iteration` or `totalIterations` or `radius0`?